businessRules/donations.js

```diff
@@ -1,54 +1,62 @@
 const { v4 } = require('uuid');
 const { createDonation } = require('../models/donations');
-const { getAllIdeas } = require('../models/ideas');
-const { getIdeas } = require('../utils/storage');
+const { findIdea } = require('../validations/ideas');
+const {
+    validateSum,
+    findIdeaWithTotalDonationSum,
+} = require('../validations/donations.js');
+const { findUser } = require('../validations/users');
 
-function createDonation(firstName, sum, userId, ideaId) {
-    let ideas = getIdeas();
-    const idea = ideas.find((idea) => ideaId === idea.id);
+function createDonationByUnregisteredUser(sum, userId, ideaId) {
+    findIdea(ideaId);
```

Comment:
You're still not doing anything with the return value from `findIdea` function call

Reply:
Yes, here I'm using it to validate the ideaId, if there is no such idea it will throw the error. 
I could create a variable, but I wont use it in this function.

Reply:
But I can create a function same as findIdea BUT that will only validate ideaId. But it's the same code is it worth it?

Reply:
> Yes, here I'm using it to validate the ideaId, if there is no such idea it will throw the error. I could create a variable, but I wont use it in this function.

If you're using it to validate the `ideaId`, then why the function is called `findIdea`? You're miscommunicating the purpose of this function in its name

Reply:
OK, I've understood. I think I've got the idea.